core/controller/src/main/scala/whisk/core/loadBalancer/LoadBalancerService.scala

```diff
@@ -84,15 +89,23 @@ trait LoadBalancer {
   def publish(action: ExecutableWhiskActionMetaData, msg: ActivationMessage)(
     implicit transid: TransactionId): Future[Future[Either[ActivationId, WhiskActivation]]]
 
+  /**
```

Comment:
^^^ (line 72) as discussed a while back in #2668, it may be worthwhile changing the LoadBalancer trait to have a generic "check()" function, which may do things other than a systemwide and per-namespace activation check. This requires some changes in RateThrottler etc.

I'd be OK waiting on this till later, but it will require changing the interface exposed by the SPI later - I'm not sure whether that is sooner vs later. WDYT @rabbah ?

Reply:
What’s the interface you have in mind? Looking at the previous proposal it was:

def check(user: Identity)(implicit tid: TransactionId): Future[Unit]

Which is fairly generic. We could include a “preprocess” method in the interface with an empty implementation in the default.

I’d have to think further about refactoring the throttle checks into the LB vs the current model where the LB exports some state and the controller uses it for throttling. I’d rather not refactor until it’s compelling and enables something new.

I also think we should accept further changes to the interface if there are many lb providers written that warrant it. 

Do you envision how this might be used in the overflow q implementation you’re working on?

Reply:
I don't think this would impact the overflow q, since the throttling is done upstream, it is just a way to make the throttling implementation specific to a loadbalancer, and possibly more comprehensive than what exists today (action specific limits, dynamic limits, etc).  

We can revisit that later as long as its ok to rev the interface (which I am ok with as well)